insights/tests/parsers/test_ssh.py

```diff
@@ -23,6 +23,40 @@
 Protocol 1
 """.strip()
 
+SSHD_CONFIG_INLINE_COMMENTS = """
+# Full line comment
+Port 22   # ssh port
+
+HostKey /etc/ssh/ssh_host_rsa_key   # main host key
+PermitRootLogin no    # disable root login
+""".strip()
+
+
+def test_sshd_config_inline_comments_preserved():
+    config = SshDConfig(context_wrap(SSHD_CONFIG_INLINE_COMMENTS))
+    assert config is not None
+
+    # Full line comment should not appear as a key
+    assert "# Full line comment" not in config
+
+    # Keys should exist
+    assert "Port" in config
+    assert "HostKey" in config
+    assert "PermitRootLogin" in config
+
+    # Inline comments should be preserved in values
+    port_value = config.get("Port")[0].value
+    assert port_value == "22   # ssh port"
```

Comment:
Sorry @1610ayushi - I just noticed this value, it includes the inline comments, which looks incorrect, and would break some exiting rules.

I think to avoid such issue, we may need to add another attribute which will returns `lines_with_comments`.

How do you think.